lib/puppet/provider/package/gem.rb

```diff
@@ -48,11 +48,12 @@ def self.provider_command
   #
   # In this case, causing the puppet_gem provider to inherit the parent gem provider's convenience gemcmd() methods, with the wrong path.
 
-  def self.execute_gem_command(command, command_options)
+  def self.execute_gem_command(command, command_options,
+                               custom_environment = {'HOME'=>ENV['HOME']})
```

Comment:
It can be problematic to define the default environment this way, because if the caller passes any env, the `HOME` default will be lost. The Puppet::Util::Execution.execute method has this problem. Maybe better to default to an empty hash `custom_environment = {}` and then in the body of the method default the `HOME` variable:

```
custom_environment['HOME'] = ENV['HOME'] unless custom_environment.key?('HOME')
```

Also you'll probably want to use `Puppet::Util.get_env` instead of `ENV` due to ruby issues on Windows.